hal/src/main/native/athena/PDP.cpp

```diff
@@ -62,77 +141,200 @@ HAL_Bool HAL_CheckPDPChannel(int32_t channel) {
   return channel < kNumPDPChannels && channel >= 0;
 }
 
-double HAL_GetPDPTemperature(int32_t module, int32_t* status) {
-  if (!checkPDPInit(module, status)) return 0;
-
-  double temperature;
+double HAL_GetPDPTemperature(HAL_PDPHandle handle, int32_t* status) {
+  PdpStatus3 pdpStatus;
+  int32_t length = 0;
+  uint64_t receivedTimestamp = 0;
 
-  *status = pdp[module]->GetTemperature(temperature);
+  HAL_ReadCANPeriodicPacket(handle, Status3, pdpStatus.data, &length,
+                            &receivedTimestamp, TimeoutMs, StatusPeriodMs,
+                            status);
 
-  return temperature;
+  return pdpStatus.bits.temp * 1.03250836957542 - 67.8564500484966;
 }
 
-double HAL_GetPDPVoltage(int32_t module, int32_t* status) {
-  if (!checkPDPInit(module, status)) return 0;
-
-  double voltage;
+double HAL_GetPDPVoltage(HAL_PDPHandle handle, int32_t* status) {
+  PdpStatus3 pdpStatus;
+  int32_t length = 0;
+  uint64_t receivedTimestamp = 0;
 
-  *status = pdp[module]->GetVoltage(voltage);
+  HAL_ReadCANPeriodicPacket(handle, Status3, pdpStatus.data, &length,
+                            &receivedTimestamp, TimeoutMs, StatusPeriodMs,
+                            status);
 
-  return voltage;
+  return pdpStatus.bits.busVoltage * 0.05 + 4.0;
```

Comment:
We should add the CTRE comment here about this magic number: `/* 50mV per unit plus 4V. */;`